migration/src/main/resources/migrations/ensure-orders-have-encounters.xml

```diff
@@ -17,28 +17,43 @@
            then create the encounters for orders that don’t have encounters
         -->
         <sql>
-            There are over 300,000 orders on both Rwink and Butaro with no linked encounter
-            select count(*) from orders where encounter_id is null
+            SET @EncTUUID :=  "a9ef19d1-8875-11ea-8c74-7a7919290ad6";
 
-            Of these, there are still between 100k and 150k on each server with no encounters on the same day
 
-            select count(*)
-            from orders o
-            left join encounter e on o.patient_id = e.patient_id and date(o.start_date) = date(e.encounter_datetime)
-            where e.encounter_id is null;
+            /* Create EncounterType */
+            INSERT INTO `encounter_type`
+            (`name`,`description`,`creator`,`date_created`,`uuid`)
+            VALUES
+            (
+            "Drug Order Encounter","for orders without encounters",1,now(),@EncTUUID);
 
-            So we need to figure out a strategy for this.
+            /* Get encounterType id */
+            select @encTId:=encounter_type_id from encounter_type where uuid=@EncTUUID;
 
-            Safest thing for now might be to create new encounters for all of these.  Later on, if we want to merge
-            some of these into other existing enocunters for the same patient on the same day, we can do so,
-            but that isn't needed for migration and might not be 100% accurate.
 
-            This could be something like a SQL script that runs the following statements:
+            /* Create encounters */
+            INSERT INTO `encounter`(`encounter_type`,`patient_id`,`encounter_datetime`,`creator`,`date_created`,`voided`,`voided_by`,`date_voided`,`void_reason`,`uuid`)
+            SELECT @encTId,patient_id,start_date,creator,date_created,voided,voided_by,date_voided,void_reason,UUID() FROM orders WHERE encounter_id is null;
+
+            /* Update orders*/
+            UPDATE orders O
+            LEFT JOIN encounter ENC ON O.patient_id = ENC.patient_id AND O.start_date=ENC.encounter_datetime AND O.creator=ENC.creator AND O.date_created=ENC.date_created
+            SET O.encounter_id=ENC.encounter_id
+            WHERE O.encounter_id is null AND ENC.encounter_type=@encTId;
+
+
+            /* Create encounter Providers */
+
+
+            INSERT INTO `encounter_provider`(`encounter_id`,`provider_id`,`encounter_role_id`,`creator`,`date_created`,`changed_by`,
+            `date_changed`,`voided`,`date_voided`,`voided_by`,`void_reason`,`uuid`)
+            SELECT enc.encounter_id,pr.provider_id,1,enc.creator,enc.date_created,enc.changed_by,
+            enc.date_changed,enc.voided,enc.date_voided,enc.voided_by,enc.void_reason,UUID()
+            FROM encounter enc
+            LEFT JOIN users US ON enc.creator = US.user_id
+            LEFT JOIN provider pr ON US.person_id=pr.person_id
+            WHERE encounter_type = @encTId;
```

Comment:
1. Should these both be inner joins?  Is there a case where we'd expect/allow these to be missing?

2. Can you clean up the formatting a bit? :)

3. Can you confirm what "encounter_provider_role = 1" is?  Are we sure this is the correct role to use on both Rwink and Butaro?

Reply:
1.yes
2. I need help :)
3.it is 1 which is "unknown" .